src/drivers/intel/dmic/dmic.c

```diff
@@ -137,69 +137,71 @@ static enum task_state dmic_work(void *data)
 static int dmic_get_hw_params(struct dai *dai,
 			      struct sof_ipc_stream_params  *params, int dir)
 {
+#if CONFIG_INTEL_DMIC_TPLG_PARAMS
 	return dmic_get_hw_params_computed(dai, params, dir);
+
+#elif CONFIG_INTEL_DMIC_NHLT
+	return dmic_get_hw_params_nhlt(dai, params, dir);
+
+#else
+	return -EINVAL;
+#endif
 }
 
-static int dmic_set_config(struct dai *dai, struct ipc_config_dai *common_config,
-			   void *spec_config)
+static int dmic_set_config(struct dai *dai, struct ipc_config_dai *common_config, void *spec_config)
 {
 	struct dmic_pdata *dmic = dai_get_drvdata(dai);
-	struct sof_ipc_dai_config *config = spec_config;
-	int32_t unmute_ramp_time_ms;
+	int32_t unmute_ramp_time_ms = LOGRAMP_TIME_MIN_MS;
```

Comment:
don't think this initialisation is needed - `unmute_ramp_time_ms` is always assigned before use.

Reply:
Done, some intermediate version needed that. No warnings seen with gcc/xt-xcc in this version.